### ISSUE TITLE
TypeError: astimezone() argument 1 must be datetime.tzinfo, not None (reprise)

### ISSUE THREAD

**Issue by mbaldessari** [geier/khal#159]
Hi Christian,

thanks a lot for fixing #158, now all of those exceptions went away ;) There is one left, likely yet another odd corner case generated by Zimbra?

I get the following with khal-0.4.0 + aeb85f69a089bcb08ae7989f96abf89a89cdf9a9

Unknown exception happened.
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/khal/khalendar/khalendar.py", line 175, in _update_vevent
    self._dbtool.update(event.raw, href=href, etag=etag)
  File "/usr/lib/python2.7/site-packages/khal/khalendar/backend.py", line 253, in update
    self._update_impl(vevent, href, etag)
  File "/usr/lib/python2.7/site-packages/khal/khalendar/backend.py", line 280, in _update_impl
    dtstartend = aux.expand(vevent, self.locale['default_timezone'], href)
  File "/usr/lib/python2.7/site-packages/khal/khalendar/aux.py", line 72, in expand
    rrule._until = rrule._until.astimezone(events_tz)
TypeError: astimezone() argument 1 must be datetime.tzinfo, not None
warning: Skipping work/273A4F5B734A4061902C5D36239D55F700000000000000000000000000000000.ics: astimezone() argument 1 must be datetime.tzinfo, not None
warning: This event will not be available in khal.

Here is the ics file causing this: http://acksyn.org/files/tmp/error-ast-rrule.ics

In the spirit of aeb85f69, maybe something like the following (just mimicking the previous fix,
I haven't dug the code much)?
```
diff --git a/khal/khalendar/aux.py b/khal/khalendar/aux.py
index e5f4d657dabd..b99a5d8c6f2c 100644
--- a/khal/khalendar/aux.py
+++ b/khal/khalendar/aux.py
@@ -69,7 +69,7 @@ def expand(vevent, default_tz, href=''):
             rrule._until = datetime(2037, 12, 31)
 
         if getattr(rrule._until, 'tzinfo', False):
-            rrule._until = rrule._until.astimezone(events_tz)
+            rrule._until = rrule._until.astimezone(events_tz or default_tz)
             rrule._until = rrule._until.replace(tzinfo=None)
 
         logger.debug('calculating recurrence dates for {0}, '
```

Thanks again for all your khal work,
Michele

**Comment by geier** [comment]
your calendar collection really is a goldmine :) I'll try to fix this tonight.

**Comment by geier** [comment]
fixed in 8d6d023891493ec

**Comment by geier** [comment]
Thanks again for reporting (and please keep reporting)!
